### ISSUE TITLE
Using HasData with DateTimeOffset causes Add-Migration to generate unneded updates

### ISSUE THREAD

**Issue by huancz** [aspnet/EntityFrameworkCore#12346]
Similar to #12194, maybe even same root cause, but I was asked to file it as a separate issue.

### Steps to reproduce

- download attached repro solution
- type "Add-Migration -name test" on PM console

Expected result: migration with empty Up() and Down() methods
Actual result:

```c#
    protected override void Up(MigrationBuilder migrationBuilder)
    {
        migrationBuilder.UpdateData(
            table: "Entity",
            keyColumn: "Id",
            keyValue: new Guid("c6a32cba-c6ea-4597-955b-2ab76025dc7d"),
            columns: new[] { "ConcurrencyStamp", "LockoutEnd" },
            values: new object[] {
                "e2115b27-76d3-432e-bac4-541c2d96e5bd",
                new DateTimeOffset(new DateTime(9999, 12, 31, 23, 59, 59, 999, DateTimeKind.Unspecified), new TimeSpan(0, 0, 0, 0, 0))
            });
    }
```

Every new migration will include the same update.

Other relevant code:
```c#
    public class Entity
    {
        public Guid Id { get; set; }
        public string ConcurrencyStamp { get; set; }
        public DateTimeOffset? LockoutEnd { get; set; }
    }

    public class ReproContext : DbContext
    {
        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);
            // commenting IsConcurrencyToken() out will stop ConcurrencyStamp updates.
            modelBuilder.Entity<Entity>()
                .Property(x => x.ConcurrencyStamp)
                .IsConcurrencyToken();
            modelBuilder.Entity<Entity>()
                .HasData(new Entity[]
                {
                    new Entity
                    {
                        Id = new Guid("c6a32cba-c6ea-4597-955b-2ab76025dc7d"),
                        ConcurrencyStamp = "e2115b27-76d3-432e-bac4-541c2d96e5bd",
                        LockoutEnd = DateTimeOffset.MaxValue
                    }
                });
        }
        // ...
    }
```

### Further technical details
EF Core version: 2.1
Database Provider: Microsoft.EntityFrameworkCore.SqlServer v2.1.0
Operating system: Windows 10 Pro
IDE: Visual Studio 2017 15.7.3

[repro-unneded-seed-updates.zip](https://github.com/aspnet/EntityFrameworkCore/files/2098052/repro-unneded-seed-updates.zip)

**Comment by AndriySvyryd** [comment]
We don't store sub millisecond values in the snapshot. As a workaround don't specify ticks in the model:
```C#
.HasData(new Entity[]
{
    new Entity
    {
        Id = new Guid("c6a32cba-c6ea-4597-955b-2ab76025dc7d"),
        ConcurrencyStamp = "e2115b27-76d3-432e-bac4-541c2d96e5bd",
        LockoutEnd = new DateTimeOffset(new DateTime(9999, 12, 31, 23, 59, 59, 999, DateTimeKind.Unspecified), new TimeSpan(0, 0, 0, 0, 0))
    }
});
```

**Comment by huancz** [comment]
Thanks for the workaround for DateTimeOffset, it should be acceptable. What about IsConcurrencyToken()? Works as intended, or will that be looked at too? Narrowing issue title to DateTimeOffset only kinda suggests you consider it to be a feature.

**Comment by AndriySvyryd** [comment]
@huancz Can you provide more details of what is the issue with `IsConcurrencyToken`?

**Comment by huancz** [comment]
@AndriySvyryd sorry, I thought it had same issue as the `DateTimeOffset` - that every migration tries to update it. Even the code I attached tries to update `ConcurrencyStamp` value in ShouldBeEmpty migration. But now I can't reproduce it anymore. If I repeat the steps exactly it only tries to update LockoutEnd, after applying your workaround further migrations are empty as they should be.

There may be something weird going on with it, but attached code doesn't reproduce it reliably. And it's low importance anyway, seeding `ConcurrencyToken` column is pretty useless concept. I wouldn't normally think of doing it, but here I didn't realize what Identity uses it for, I just took all column values from DB without really thinking about the content.

New title is correct, after fix for `DateTimeOffset` lands, this issue can be considered as completed.
